### ISSUE TITLE
error: argument 2 of '__atomic_load' discards 'volatile' qualifier

### ISSUE THREAD

**Issue by ggardet** [nnstreamer/nnstreamer#3575]
# Issue Description
Build of v2.0.0 fails on openSUSE Tumbleweed on `__atomic_load`:
```
[   45s] FAILED: gst/nnstreamer/libnnstreamer.so.p/_home_abuild_rpmbuild_BUILD_nnstreamer-2.0.0_gst_nnstreamer_tensor_filter_tensor_filter_common.c.o 
[   45s] cc -Igst/nnstreamer/libnnstreamer.so.p -Igst/nnstreamer -I../gst/nnstreamer -Igst/nnstreamer/include -I../gst/nnstreamer/include -Igst/nnstreamer/tensor_transform -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/gstreamer-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/orc-0.4 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu89 '-DVERSION="2.0.0"' '-DVERSION_MAJOR="2"' '-DVERSION_MINOR="0"' '-DVERSION_MICRO="0"' -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Waggregate-return -Wold-style-definition -Wdeclaration-after-statement '-DNNSTREAMER_CONF_FILE="/etc/nnstreamer.ini"' -DGLIB_USE_G_MEMDUP2 -DENABLE_PROTOBUF=1 -DENABLE_TENSORFLOW=1 -DENABLE_TENSORFLOW2_LITE=1 -DHAVE_ORC=1 -Wredundant-decls -fPIC -pthread -MD -MQ gst/nnstreamer/libnnstreamer.so.p/_home_abuild_rpmbuild_BUILD_nnstreamer-2.0.0_gst_nnstreamer_tensor_filter_tensor_filter_common.c.o -MF gst/nnstreamer/libnnstreamer.so.p/_home_abuild_rpmbuild_BUILD_nnstreamer-2.0.0_gst_nnstreamer_tensor_filter_tensor_filter_common.c.o.d -o gst/nnstreamer/libnnstreamer.so.p/_home_abuild_rpmbuild_BUILD_nnstreamer-2.0.0_gst_nnstreamer_tensor_filter_tensor_filter_common.c.o -c /home/abuild/rpmbuild/BUILD/nnstreamer-2.0.0/gst/nnstreamer/tensor_filter/tensor_filter_common.c
[   45s] In file included from /usr/include/glib-2.0/glib/gthread.h:32,
[   45s]                  from /usr/include/glib-2.0/glib/gasyncqueue.h:32,
[   45s]                  from /usr/include/glib-2.0/glib.h:32,
[   45s]                  from ../gst/nnstreamer/hw_accel.h:19,
[   45s]                  from /home/abuild/rpmbuild/BUILD/nnstreamer-2.0.0/gst/nnstreamer/tensor_filter/tensor_filter_common.c:28:
[   45s] /home/abuild/rpmbuild/BUILD/nnstreamer-2.0.0/gst/nnstreamer/tensor_filter/tensor_filter_common.c: In function 'accl_hw_get_type':
[   45s] /usr/include/glib-2.0/glib/gatomic.h:113:5: error: argument 2 of '__atomic_load' discards 'volatile' qualifier [-Werror=incompatible-pointer-types]
[   45s]   113 |     __atomic_load (gapg_temp_atomic, &gapg_temp_newval, __ATOMIC_SEQ_CST); \
[   45s]       |     ^~~~~~~~~~~~~
[   45s] /usr/include/glib-2.0/glib/gthread.h:260:7: note: in expansion of macro 'g_atomic_pointer_get'
[   45s]   260 |     (!g_atomic_pointer_get (location) &&                             \
[   45s]       |       ^~~~~~~~~~~~~~~~~~~~
[   45s] /home/abuild/rpmbuild/BUILD/nnstreamer-2.0.0/gst/nnstreamer/tensor_filter/tensor_filter_common.c:2767:7: note: in expansion of macro 'g_once_init_enter'
[   45s]  2767 |   if (g_once_init_enter (&g_accl_hw_type_id__volatile)) {
[   45s]       |       ^~~~~~~~~~~~~~~~~
[   45s] cc1: all warnings being treated as errors
```

Expected Result
============
Build should succeed.

How to Reproduce
===============
Try to build with GCC11.

Further Information
===============
* x86_64: https://build.opensuse.org/package/live_build_log/home:Guillaume_G:branches:science:machinelearning/nnstreamer/openSUSE_Tumbleweed/x86_64
* aarch64: https://build.opensuse.org/package/live_build_log/home:Guillaume_G:branches:science:machinelearning/nnstreamer/openSUSE_Tumbleweed/aarch64

**Comment by anyj0527** [comment]
It seems that https://github.com/nnstreamer/nnstreamer/pull/3532/commits/fd48e290a0dbdb805f7fab285c3a2d849933b052
should be applied to v2.0

**Comment by myungjoo** [comment]
> It seems that [fd48e29](https://github.com/nnstreamer/nnstreamer/commit/fd48e290a0dbdb805f7fab285c3a2d849933b052) should be applied to v2.0

You may add hotfixes to lts/2.0.y

**Comment by ggardet** [comment]
fd48e29 indeed fixes this error. Then, I have another error: https://github.com/nnstreamer/nnstreamer/issues/3579

**Comment by anyj0527** [comment]
Merged #3580 should resolve this issue and #3579. Please check the updated lts/2.0 branch

**Comment by ggardet** [comment]
> Merged #3580 should resolve this issue and #3579. Please check the updated lts/2.0 branch

Indeed, it builds properly now! Thanks!
